### ISSUE TITLE
Remove support for "att" and "ucb" universes (i.e., OS personalities)

### ISSUE THREAD

**Issue by krader1961** [att/ast#81]
One way to make it easier to switch to a new build system (e.g., Cmake or Meson) is to eliminate support for long dead operating systems. Such as DYNIX/3 from Sequent Computer Systems. As evidenced by these statements emitted by `bin/package make`:

```
iffe: test: is universe a command ... no
iffe: test: is getuniverse a library function ... no
iffe: test: is sys/universe.h a header ... no
```

See my [comment](https://github.com/att/ast/issues/42#issuecomment-339210893) on issue #42.

**Comment by siteshwar** [comment]
I would be mostly interested in supporting just Linux and BSD family of operating systems. I am fine with dropping support for rest.

**Comment by krader1961** [comment]
I think we should try to support as many operating systems as possible that are still being actively maintained. Not just used but being maintained by which I mean that bugs are being fixed in a timely manner. There may be people still using Solaris for example but it is dead. From its wikipedia page:

> On September 2, 2017, Simon Phipps reported that Oracle had laid off the Solaris core development staff, interpreting it as sign that Oracle no longer intends to support future development of the platform.

Clearly we shouldn't expend resources trying to make ksh93 work on dead OS's. Removing support for an OS that is supported by ksh93u is a more contentious question and probably needs to be handled on a case by case basis. I'm comfortable removing support for DYNIX/3 because it's dual universe support adds non-trivial complexity to ksh and no one is going to care that we no longer support it. The same is true for UWIN. Not to mention pre-ANSI C compilers.

**Comment by krader1961** [comment]
FWIW, It turns out that the att/ucb universe mechanism was found in OS's other than DYNIX/3: https://en.wikipedia.org/wiki/Universe_(Unix). But as all of those operating systems are now dead we can still safely remove support for dual universe OS's. Also, to the extent there are any SVR4 based distros being used that still have a /usr/ucb directory the commands therein should not be special-cased by ksh93. The native SVR4 variants are perfectly acceptable.

**Comment by krader1961** [comment]
Removing support for att versus ucb "universe"s is going to be really
difficult without breaking things. At least inadvertantly. Consider this
block of code from function `B_echo()` in *src/cmd/ksh93/bltins/print.c*:

```
       static char bsd_univ;

       /* This mess is because /bin/echo on BSD is different */
       if(!prdata.sh->universe)
       {
               register char *universe;
               if(universe=astconf("UNIVERSE",0,0))
                       bsd_univ = (strcmp(universe,"ucb")==0);
               prdata.sh->universe = 1;
       }
       if(!bsd_univ)
               return(b_print(0,argv,(Shbltin_t*)&prdata));
```

Why does the builtin `echo` behavior depend on the current "universe"?
The whole point of builtins is to avoid this type of OS specific behavior
so that portable scripts are easier to write. There are even worse examples
of this such as the `settime()` function in *src/lib/libcmd/date.c*. And
what the hell does `!prdata.sh->universe` mean in the above block of
code? Too, that is not even close to the most hard to understand block
of code involving dual-universe support.

Also, declaring `bsd_univ` to be static is just plain gross. That is
relying on static ints being initialized to zero. Furthermore, the
"static"ness of the value isn't needed. It should be declared like this:

```
    char bsd_univ = 0;
```

Bottom line is this cleanup can't be done until the unit tests pass without
failure on current Linux and BSD based distros.

**Comment by catull** [comment]
These constants are candidates: _UWIN, __EMX__ (remember OS/2?) and __INTERIX.
Created PR #286 to address this.

**Comment by colstrom** [comment]
In response to:

> There may be people still using Solaris for example but it is dead. 

While Solaris proper may be dead, illumos (https://github.com/illumos/illumos-gate) and derivatives are still actively developed and used in new deployments.

Incidentally, `/usr/ucb` is still a thing on those systems. Aside from `/usr/ucb/install`, I can't recall needing much from there, though.

**Comment by krader1961** [comment]
I'm closing this. See issue #370 for why. Too many bad design choices have been included in this project that a lot of people consider unquestionable for my taste. So I give up.
